backend/ohq/permissions.py

```diff
@@ -392,3 +395,86 @@ def has_permission(self, request, view):
         # Head TAs+ can make changes
         if view.action in ["create", "destroy", "update", "partial_update"]:
             return membership.is_leadership
+
+
+class EventPermission(permissions.BasePermission):
+    def get_membership_from_event(self, request, event):
+        event_course_relation = EventRelation.objects.filter(event=event).first()
+        membership = Membership.objects.filter(
+            course_id=event_course_relation.object_id, user=request.user
+        ).first()
+        return membership
+
+    def has_object_permission(self, request, view, obj):
+        return self.has_permission(request, view)
+
+    def has_permission(self, request, view):
+        # Anonymous users can't do anything
+        if not request.user.is_authenticated:
+            return False
+
+        if view.action in ["create"]:
+            course_pk = json.loads(request.body)["courseId"]
+            course = Course.objects.get(pk=course_pk)
+            membership = Membership.objects.filter(course=course, user=request.user).first()
+            if membership is None:
+                return False
+            return membership.is_ta
+
+        if view.action in ["partial_update", "update"]:
```

Comment:
Move these update permissions and the retrieve and destroy permissions to has_object_permissions

Reply:
Also, might want to associate event with a TA, and only allow the specific TA who created the event, head TA's, and prof. to edit/delete the event